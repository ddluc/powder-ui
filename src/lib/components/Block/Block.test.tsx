import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Block, BlockProps } from './index';

afterEach(cleanup);

const args: BlockProps = {
  display: 'block',
  height: '100px',
  width: '100px',
  position: 'absolute',
  left: '2px',
  right: '3px',
  top: '1px',
  bottom: '4px',
  margin: ['10px', '20px', '5px', '3px'],
  padding: ['3px', '12px', '6px', '9px'],
  opacity: 1,
  border: 'solid 1px #EDEDED',
  background: {
    color: 'red',
    position: '50% 50%',
    image: 'inherit',
    repeat: 'no-repeat',
    size: 'contain'
  },
  outline: 'none',
  textAlign: 'center',
  overflow: 'hidden',
  zIndex: 1,
  float: 'right'
};

// Define unit tests
describe('Block', () => {
  it('should render with defaults', () => {
    const result = render(<Block />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });
  it('should render with defined props', () => {
    const result = render(<Block {...args} />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });
});
