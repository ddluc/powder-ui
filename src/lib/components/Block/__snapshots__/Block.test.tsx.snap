// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block should render with defaults 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

<div
  class="c0"
/>
`;

exports[`Block should render with defined props 1`] = `
.c0 {
  display: block;
  height: 100px;
  width: 100px;
  position: absolute;
  left: 2px;
  right: 3px;
  left: 1px;
  bottom: 4px;
  margin: 10px 20px 5px 3px;
  padding: 3px 12px 6px 9px;
  opacity: 1;
  border: solid 1px #EDEDED;
  background-color: red;
  background-position: 50% 50%;
  background-image: inherit;
  background-repeat: no-repeat;
  background-size: contain;
  outline: none;
  text-align: center;
  overflow: hidden;
  zIndex: 1;
  float: right;
}

<div
  class="c0"
  display="block"
  height="100px"
  opacity="1"
  overflow="hidden"
  width="100px"
/>
`;
